metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/HiveConnectorFastPartitionService.java

```diff
@@ -389,9 +391,17 @@ public void deletePartitions(
             .collect(Collectors.toList());
 
         if (sort != null) {
-            //it can only support sortBy partition Name
-            final Comparator<PartitionInfo> nameComparator = Comparator.comparing(p -> p.getName().toString());
-            ConnectorUtils.sort(filteredPartitionList, sort, nameComparator);
+            if (sort.hasSort() && sort.getSortBy().equalsIgnoreCase(SORT_BY_DATE_CREATED)) {
+                final Comparator<PartitionInfo> dateCreatedComparator = Comparator.comparing(
+                    p -> p.getAudit().getCreatedDate(),
+                    Comparator.nullsLast(Date::compareTo));
```

Comment:
Can you please check the p.getAudit() will not have a null pointer?

Reply:
Updated.